### ISSUE TITLE
Add timeout handling

### ISSUE THREAD

**Issue by natalia-osa** [TimOliver/TOSMBClient#25]
When you create and resume `TOSMBSessionDownloadTask` it may work infinitely if the network connection was lost. It would be good to add the timeout, after which the task would fail.

**Comment by narendra-ct** [comment]
Yes, Im also having the same issue while downloading if network lost its trying to connect continuously. It would be great if introduce any timeout/network fail conditions.
Thanks,
Narendra

**Comment by TimOliver** [comment]
This sadly needs to be added to `libdsm` in order to be exposed here. I [did some research](http://stackoverflow.com/questions/25665837/is-it-possible-to-activate-tcp-keepalive-on-apple-ios-devices) and had a play on how to configure the sockets API to achieve this, but I couldn't get it working.

**Comment by natalia-osa** [comment]
Thank you for the interest and the answer.
